### ISSUE TITLE
Clojure's DynamicClassLoader classes cannot be instrumented

### ISSUE THREAD

**Issue by rymndhng** [DataDog/dd-trace-java#454]
We're seeing quite a few of these debug messages.

```
[qtp1841695610-94-b15922e4-b495-4d81-9d44-c621c6fd997e] DEBUG datadog.trace.agent.tooling.AgentInstaller$LoggingListener - Failed to handle compile__stub.hugsql.adapter.clojure_java_jdbc.HugsqlAdapterClojureJavaJdbc for transformation on classloader clojure.lang.DynamicClassLoader@557e42a4: Cannot resolve type description for hugsql.adapter.HugsqlAdapter
[qtp1841695610-94-b15922e4-b495-4d81-9d44-c621c6fd997e] DEBUG datadog.trace.agent.tooling.AgentInstaller$LoggingListener - Failed to handle hugsql.adapter.clojure_java_jdbc.HugsqlAdapterClojureJavaJdbc for transformation on classloader clojure.lang.DynamicClassLoader@557e42a4: Cannot resolve type description for hugsql.adapter.HugsqlAdapter
```

From experience with NewRelic, we've opted to blacklist the classloader. See http://corfield.org/blog/2016/07/29/clojure-new-relic-slow-startup/ 

Is this an option we can potentially support?

**Comment by rymndhng** [comment]
On second thought this may not be desirable. We have a jetty handler from Clojure's Classloader that was instrumented properly.

**Comment by realark** [comment]
Thanks for reporting that @rymndhng. I guess there can be useful classes to instrument in Clojure's dynamic loader. Good to know.

Generally speaking, having an option to skip user-configured classloaders is something we could add. If this is still something you want to do feel free to submit a PR. Or you can open another issue asking for the feature and we'll prioritize it (disclaimer: it may be a little while before we can work on it).
